src/status_im2/subs/chat/messages.cljs

```diff
@@ -96,63 +97,81 @@
+        (swap! all-messages conj message)
+        (when (and album-id (> (count (get @albums album-id)) 3))
+          (swap! all-messages (fn [data] (filterv #(not= album-id (:album-id %)) data)))
+          (swap! all-messages conj {:album        (get @albums album-id)
+                                    :album-id     album-id
+                                    :message-id   album-id
+                                    :content-type status-im.constants/content-type-album}))))
+    @all-messages))
 
 (re-frame/reg-sub
- :chats/raw-chat-messages-stream
- (fn [[_ chat-id] _]
-   [(re-frame/subscribe [:chats/message-list chat-id])
-    (re-frame/subscribe [:chats/chat-messages chat-id])
-    (re-frame/subscribe [:chats/pinned chat-id])
-    (re-frame/subscribe [:chats/loading-messages? chat-id])
-    (re-frame/subscribe [:chats/synced-from chat-id])
-    (re-frame/subscribe [:chats/chat-type chat-id])
-    (re-frame/subscribe [:chats/joined chat-id])])
- (fn [[message-list messages pin-messages loading-messages? synced-from chat-type joined] [_ chat-id]]
-   ;;TODO (perf)
-   (let [message-list-seq (models.message-list/->seq message-list)]
-     ; Don't show gaps if that's the case as we are still loading messages
-     (if (and (empty? message-list-seq) loading-messages?)
-       []
-       (-> message-list-seq
-           (chat.db/add-datemarks)
-           (hydrate-messages messages pin-messages)
-           (chat.db/collapse-gaps chat-id
-                                  synced-from
-                                  (datetime/timestamp)
-                                  chat-type
-                                  joined
-                                  loading-messages?))))))
+  :chats/raw-chat-messages-stream
+  (fn [[_ chat-id] _]
+    [(re-frame/subscribe [:chats/message-list chat-id])
+     (re-frame/subscribe [:chats/chat-messages chat-id])
+     (re-frame/subscribe [:chats/pinned chat-id])
+     (re-frame/subscribe [:chats/loading-messages? chat-id])
+     (re-frame/subscribe [:chats/synced-from chat-id])
+     (re-frame/subscribe [:chats/chat-type chat-id])
+     (re-frame/subscribe [:chats/joined chat-id])])
+  (fn [[message-list messages pin-messages loading-messages? synced-from chat-type joined] [_ chat-id]]
+    ;;TODO (perf)
+    (let [message-list-seq (models.message-list/->seq message-list)]
+      ; Don't show gaps if that's the case as we are still loading messages
+      (if (and (empty? message-list-seq) loading-messages?)
+        []
+        (-> message-list-seq
+            (chat.db/add-datemarks)
+            (hydrate-messages messages pin-messages)
+            (chat.db/collapse-gaps chat-id
+                                   synced-from
+                                   (datetime/timestamp)
+                                   chat-type
+                                   joined
+                                   loading-messages?)
+            (find-albums))))))
```

Comment:
does this not change the previous implementation of this sub? Should you not create a new sub for this behaviour? 🤔

Reply:
It is intended to be changed. When need to always "find albums" when loading new messages.